AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/test/TestGetKeys.dfy

```diff
@@ -305,162 +247,250 @@ module TestGetKeys {
 
-    var hv2versionResult :- expect keyStore.GetBranchKeyVersion(
-      Types.GetBranchKeyVersionInput(
-        branchKeyIdentifier := hv2BranchKeyId,
-        branchKeyVersion := hv2BranchKeyIdWithEC
-      ));
+    GetBranchKeyVersionWithIncorrectKmsKeyArnHelper(keyStore, branchKeyId, branchKeyIdActiveVersion);
+    GetBranchKeyVersionWithIncorrectKmsKeyArnHelper(keyStore, hv2BranchKeyId, hv2BranchKeyVersion);
+  }
+
+  method {:test} TestGetActiveKeyWrongLogicalKeyStoreName() {
+    var kmsClient :- expect KMS.KMSClient();
+    var ddbClient :- expect DDB.DynamoDBClient();
 
-    var hv2testBytes :- expect UTF8.Encode(hv2BranchKeyIdWithEC);
+    var keyStore :- expect DefaultKeyStore(kmsId:=keyArn, physicalName := branchKeyStoreName, logicalName := incorrectLogicalName, ddbClient? := Some(ddbClient), kmsClient? := Some(kmsClient));
+    GetActiveKeyWrongLogicalKeyStoreName(keyStore, branchKeyId, "1");
+    GetActiveKeyWrongLogicalKeyStoreName(keyStore, hv2BranchKeyId, "2");
 
-    expect hv2versionResult.branchKeyMaterials.branchKeyIdentifier == hv2BranchKeyId;
-    expect hv2versionResult.branchKeyMaterials.branchKeyVersion == hv2BranchKeyIdActiveVersionUtf8Bytes == hv2testBytes;
-    expect |hv2versionResult.branchKeyMaterials.branchKey| == 32;
-  }
+    GetBeaconKeyWrongLogicalKeyStoreName(keyStore, branchKeyId, "1");
+    GetBeaconKeyWrongLogicalKeyStoreName(keyStore, hv2BranchKeyId, "2");
 
+    GetVersionKeyWrongLogicalKeyStoreName(keyStore, branchKeyId, branchKeyIdActiveVersion, "1");
+    GetVersionKeyWrongLogicalKeyStoreName(keyStore, hv2BranchKeyId, hv2BranchKeyVersion, "2");
+  }
 
-  method {:test} TestGetActiveKeyWithIncorrectKmsKeyArn() {
+  method {:test} TestGetKeyDoesNotExistFails()
+  {
     var kmsClient :- expect KMS.KMSClient();
     var ddbClient :- expect DDB.DynamoDBClient();
+
+    var keyStore :- expect DefaultKeyStore(kmsId := keyArn, physicalName := branchKeyStoreName, logicalName := logicalKeyStoreName, ddbClient? := Some(ddbClient), kmsClient? := Some(kmsClient));
+    GetActiveKeyDoesNotExistFailsHelper(keyStore, "Robbie");
+    GetBeaconKeyDoesNotExistFailsHelper(keyStore, "Robbie");
+    GetBranchKeyVersionDoesNotExistFailsHelper(keyStore, "Robbie", branchKeyIdActiveVersion);
+  }
+
+  method {:test} TestGetKeysWithNoClients() {
     var kmsConfig := Types.KMSConfiguration.kmsKeyArn(keyArn);
 
-    var keyStoreConfig := Types.KeyStoreConfig(
-      id := None,
-      kmsConfiguration := kmsConfig,
-      logicalKeyStoreName := logicalKeyStoreName,
-      storage := Some(
-        Types.ddb(
-          Types.DynamoDBTable(
-            ddbTableName := branchKeyStoreName,
-            ddbClient := Some(ddbClient)
-          ))),
-      keyManagement := Some(
-        Types.kms(
-          Types.AwsKms(
-            kmsClient := Some(kmsClient)
-          )))
-    );
+    var keyStore :- expect DefaultKeyStore(kmsId:=keyArn, physicalName:=branchKeyStoreName, logicalName := logicalKeyStoreName);
```

Comment:
Blocking: Unfortunately, this is not testing the intended behavior.
This uses `Fixtures#DefaultKeyStore` to populate the clients;
[but this test asserts the behavior of `Index#KeyStore`](https://github.com/aws/aws-cryptographic-material-providers-library/blob/hv-2/hv-2/AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/src/Index.dfy#L41).

For this particular test, we MUST NOT use the `Fixture` method.